ui/lib/core/addon/utils/sanitize-path.js

```diff
@@ -4,10 +4,40 @@
  */
 
 export function sanitizePath(path) {
+  if (!path) return '';
   //remove whitespace + remove trailing and leading slashes
   return path.trim().replace(/^\/+|\/+$/g, '');
 }
 
 export function ensureTrailingSlash(path) {
   return path.replace(/(\w+[^/]$)/g, '$1/');
 }
+
+/**
+ * getRelativePath is for removing matching segments of a subpath from the front of a full path
+ * @param {string} fullPath eg apps/prod/app_1/test
+ * @param {string} rootPath eg apps/prod
+ * @returns the leftover segment, eg app_1/test
+ */
+export function getRelativePath(fullPath = '', rootPath = '') {
+  const root = sanitizePath(rootPath);
+  const full = sanitizePath(fullPath);
+
+  if (!root) {
+    return full;
+  }
+
+  return root
+    .split('/')
+    .reduce(
+      (prev, curr) => {
+        if (prev[0] === curr) {
+          // if they match, shift from the front
+          prev.shift();
+        }
+        return prev;
+      },
+      [...full.split('/')]
+    )
+    .join('/');
+}
```

Comment:
I'm wondering if this could be simplified to use `split` on the full path with the root as the characters to match instead?
```js
return fullPath.split(rootPath)[1].slice(1); 
```
Given the example in the comment this would return `app_1/test`. I'm probably missing context here so feel free to disregard but wanted to point it out.

Reply:
The root path might be multiple segments if the user's root namespace is something like `admin/child` -- that's why this method is rather dynamic. But happy to entertain other simpler ways of doing this!

Reply:
Oh wait -- i understand your suggestion now. I'll give that a shot!